src/Struct/FindologicProduct.php

```diff
@@ -762,14 +762,30 @@ protected function setOrdernumberByProduct(ProductEntity $product): void
     protected function setCategoriesAndCatUrls(): void
     {
         $productCategories = $this->product->getCategories();
-        if ($productCategories === null || empty($productCategories->count())) {
+        $productVariants = $this->product->getChildren();
+
+        // If there are no product categories, and no children either, we throw ProductHasNoCategoriesException.
+        if ($productCategories->count() === 0 && $productVariants->count() === 0) {
```

Comment:
You should count the categories of the variants, not the variants themselves. Right now this could cause an issue when a product has no categories, but at least one variant. It would pass, but even tho neither the parent nor any of the variants has a category assigned. Shouldn't there be already a test for this? If not, please create one, otherwise please let me know the name, so I can check what went wrong here.